students/luftsmeerflier/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,134 @@
+#/usr/bin/env python3
+import sys
+
+donors = {
+	"John Quigley": [
+		("2019-02-04", 5000.00),
+		("2019-02-04", 4250.00),
+		("2019-02-04", 3000.00)
+	],
+	"Sara Smith": [
+		("2020-04-02", 2000.00),
+		("2019-02-04", 42000.00),
+		("2019-02-04", 32000.00)
+	],
+	"Jacob van der Schmidt": [			
+		("2012-04-20", 20.00),
+		("2019-02-04", 2000.00),
+		("2019-02-04", 3400.00)
+	],
+	"Ogden Nash": [
+		("2019-02-04", 5000000.00),
+		("2019-02-04", 500000.00),
+		("2019-02-04", 50000.00)
+	]
+}
+
+def send_thank_you(first_name='', last_name=''):
+	names_list = list(donors.keys())
+
+	_list = input("Type `list` for a list of donors or press the carriage key to skip\n")
+	if _list == "list":
+		for donor in names_list:
+			print(donor)
+		print('\n')
+	else:
+		pass
+
+	name = input('What is the name of the donor?\n').title()
+
+	if name in names_list:
+		index = names_list.index(name)
+		name = names_list[index]
+		donor = donors[name]
+		date = donor[0][0]
+		amount = donor[0][1]
+		write_letters(name, amount, date)
+	
+
+def write_letters(name, amount, date):	
+
+	text = "Dear {},\n\n    Thank you for your donation in the amount of {} on {}. Please keep this letter for your records.\n\n"
+
+	letterhead = "{}.txt".format(name).replace(" ", "_")
+	# open a letter template
+	formatted = text.format(name, amount, date)
+	thank_you = open("{}".format(letterhead), "w+")
+	thank_you.write(formatted)
+	thank_you.close()
+	print('\n')
+	print("Thank you letter has been created")
+	print('\n')
+
+
+def create_report():
+	#print(donor_list[0].ljust(20) + donor_list[1].ljust(10) + donor_list[2].ljust(20) + donor_list[3].ljust(8))
+	# print(donor_list[0].ljust(20))
+	print('\n')
+	print("Donor Name".ljust(28) + '|'.ljust(2) + "Total Given" + '|'.rjust(2) + "Num Gifts".rjust(10) + "|".ljust(2) + "Average Gift")
+	print('-' * 70)
+
+	names_list = list(donors.keys())
+	names_list.sort()
```

Comment:
the sorting functionality isn't working as expected - needs to be sorted by total given.